src/sage/schemes/curves/plane_curve_arrangement.py

```diff
@@ -0,0 +1,1259 @@
+        Return the curve arrangement obtained by removing ``curves``.
+
+        INPUT:
+
+        - ``curves`` -- a curve or curve arrangement
+
+        OUTPUT:
+
+        A new curve arrangement with the given curve(s)
+        ``h`` removed.
+
+        EXAMPLES::
+
+            sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: h = H([x * y, x + y + 1, x^3 - y^5, x^2 * y^2 + x^5 + y^5, (x^2 + y^2)^3 + (x^3 + y^3 - 1)^2])
+            sage: C = h[-1]
+            sage: h.deletion(C)
+            Arrangement (x*y, x + y + 1, -y^5 + x^3, x^5 + y^5 + x^2*y^2)
+            in Affine Space of dimension 2 over Rational Field
+            sage: h.deletion(x)
+            Traceback (most recent call last):
+            ...
+            ValueError: curve is not in the arrangement
+            """
+        parent = self.parent()
+        curves = parent(curves)
+        planes = list(self)
+        for curve in curves:
+            try:
+                planes.remove(curve)
+            except ValueError:
+                raise ValueError('curve is not in the arrangement')
+        return parent(planes)
+
+    def change_ring(self, base_ring):
+        """
+        Return curve arrangement over the new base ring.
+
+        INPUT:
+
+        - ``base_ring`` -- the new base ring; must be a field for
+          curve arrangements
+
+        OUTPUT:
+
+        The curve arrangement obtained by changing the base
+        field, as a new curve arrangement.
+
+        EXAMPLES::
+
+            sage: # needs sage.rings.number_field
+            sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: A = H(y^2 - x^3, x, y, y^2 + x * y + x^2)
+            sage: K.<a> = CyclotomicField(3)
+            sage: A.change_ring(K)
+            Arrangement (-x^3 + y^2, x, y, x^2 + x*y + y^2) in Affine Space of
+            dimension 2 over Cyclotomic Field of order 3 and degree 2
+        """
+        parent = self.parent().change_ring(base_ring)
+        curves = tuple(c.change_ring(base_ring) for c in self)
+        return parent(curves)
+
+    def coordinate_ring(self):
+        """
+        Return the coordinate ring of ``self``.
+
+        OUTPUT:
+
+        The coordinate ring of the curve arrangement.
+
+        EXAMPLES::
+
+            sage: L.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: C = L(x, y)
+            sage: C.coordinate_ring()
+            Multivariate Polynomial Ring in x, y over Rational Field
+        """
+        return self.curves()[0].defining_polynomial().parent()
+
+    def defining_polynomials(self):
+        r"""
+        Return the defining polynomials of the elements of ``self``.
+
+        EXAMPLES::
+
+            sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: A = H(y^2 - x^3, x, y, y^2 + x * y + x^2)
+            sage: A.defining_polynomials()
+            (-x^3 + y^2, x, y, x^2 + x*y + y^2)
+        """
+        return tuple(h.defining_polynomial() for h in self)
```

Comment:
```suggestion
        return tuple([h.defining_polynomial() for h in self._curves])
```
This is faster (including the list part surprisingly) and the explicitness is useful too.